cupy/_statistics/correlation.py

```diff
@@ -131,13 +145,62 @@ def cov(a, y=None, rowvar=True, bias=False, ddof=None):
     if ddof is None:
         ddof = 0 if bias else 1
 
-    fact = X.shape[1] - ddof
+    w = None
+    if fweights is not None:
+        fweights = cupy.asarray(fweights, dtype=float)
```

Comment:
Let's add some typechecks here, fweights should only be `int` or `cupy.ndarray`.
Other types are forbidden.

Reply:
There is a typecheck in the next line though, `if not cupy.all(fweights == cupy.around(fweights)):`, which I will replace with `fweights.dtype.char in 'bBhHiIlLqQ'`. Is that sufficient for typecheck or should there be something before line 150 here?

Reply:
Although now that I think about it, line 150 sets the dtype to float, so `fweights.dtype.char in 'bBhHiIlLqQ'` would not work after that anymore. How can I best structure this?

Reply:
do the check before, and then with `astype` cast it to float if needed. Probably the numpy casting rules will do it for you if you use it later.